File: sample/src/main/java/com/kandoka/springframework/beans/factory/HierarchicalBeanFactory.java
package com.kandoka.springframework.beans.factory;

import com.kandoka.springframework.beans.BeansException;
import com.kandoka.springframework.beans.factory.config.AutowireCapableBeanFactory;
import com.kandoka.springframework.beans.factory.config.BeanDefinition;
import com.kandoka.springframework.beans.factory.config.ConfigurableBeanFactory;

/**
 * @Description 它提供了可以获取父类 BeanFactory
 *              方法，属于是一种扩展工厂的层次子接口。
 * @Author handong3
 * @Date 2022/2/16 16:38
 */
public interface HierarchicalBeanFactory extends BeanFactory {

}

File: core/src/main/java/com/kandoka/core/beans/factory/HierarchicalBeanFactory.kt
package com.kandoka.core.beans.factory

/**
 *
 *  @Description Sub-interface implemented by bean factories that can be part
 * of a hierarchy.
 *
 *
 * The corresponding `setParentBeanFactory` method for bean
 * factories that allow setting the parent in a configurable
 * fashion can be found in the ConfigurableBeanFactory interface.
 *
 *  @Author handong3
 *  @Date 2022/11/9 18:13
 *
 */
interface HierarchicalBeanFactory: BeanFactory {
}